File: Python3/Buy Two Chocolates.py
#1
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        prices.sort()
        price = prices[0] + prices[1]
        if price <= money:
            return money - price
        return money

#2
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        first_min = second_min = maxsize

        for p in prices:
            if p < first_min:
                second_min, first_min = first_min, p
            elif p < second_min:
                second_min = p

        min_price = first_min + second_min
        return money - min_price if min_price <= money else money

File: Golang/Buy Two Chocolates.go
//1
func buyChoco(prices []int, money int) int {
    sort.Ints(prices)
    min := prices[0] + prices[1]

    if money >= min {
        return money - min
    }
    
    return money
}

//2
func buyChoco(prices []int, money int) int {
    firstMin := 100
    secondMin := 100

    for _, price := range prices {
        if price < firstMin {
            secondMin = firstMin
            firstMin = price
        } else if price < secondMin {
            secondMin = price
        }
    }

    min := firstMin + secondMin

    if money - min >= 0 {
        return money - min
    } 

    return money
}